crates/ra_hir/src/code_model.rs

```diff
@@ -989,6 +989,12 @@ impl TypeParam {
             ty: InEnvironment { value: ty, environment },
         }
     }
+
+    pub fn default(self, db: &dyn HirDatabase) -> Option<Ty> {
+        let params = db.generic_defaults(self.id.parent);
+        let local_idx: u32 = self.id.local_id.into_raw().into();
+        params.get(local_idx as usize).map(|d| d.clone())
```

Comment:
hm yeah `local_idx` isn't necessarily the correct index here. There's a function that will give you the correct index in `ra_hir_ty::utils`, but it's currently private to that crate: `generics(db, parent_def).param_idx(param_id)`. let's maybe just export that as a function from ra_hir_ty? i.e.
```rust
pub fn param_idx(db, param: TypeParamId) -> Option<usize> { generics(db, param.parent).param_idx(param) }
```

Reply:
also, `.map(|d| d.clone())` = `.cloned()`, I think

Reply:
oh, and this function should wrap the `Ty` in a `code_model::Type` :grimacing: